### DISCUSSION TITLE
Define scalar product of polarisation vectors

### DISCUSSION THREAD

**Discussion by roberpsm** [FeynCalc/feyncalc#96] [Q&A] [answered]
Hello,

I'm trying to set the kinematics of a 2->2 vector boson scattering at amplitude level (no polarisation sum since i am not interested in the square amplitude). 
I've defined the scalar products of external momenta via ScalarProduct[p1,p1]=mV^2, ScalarProduct[p1,p2]=(s-2mV^2)/2, etc and FC works fine.
My problem is in the definition of scalar products between the polarisation vectors (others than the corresponding ones to transversality), for example ScalarProduct[p1,PolarizationVector[p2]]. Maybe I am confusing with the syntax here, I was trying different ways but no one works... 

Please see the attached file for a short example (one-loop ZZ->ZZ in the SM through a bubble of Higgs boson): the products momenta.momenta are well defined in the argument of the B0 functions but the products polarisation.polarisation are still there (no products momenta.polarisation are present in this example but I am interested on them too).

[Hloop-ZZtoZZ.nb.zip](https://github.com/FeynCalc/feyncalc/files/5881037/Hloop-ZZtoZZ.nb.zip)

I'm using Mathematica 11.3.0, FeynCalc 9.3.1 and FeynArts 3.11.

Thank you in advance!

**Comment by magnunm** [comment, marked as answer]
Firstly you have forgotten that the 3 and 4 polarizations are outgoing and therefore complex conjugated. The scalar product declarations should have
```Mathematica
ComplexConjugate[PolarizationVector[p4]]
```
and same for `p3`.

Using the complex conjugated polarization vectors I still could not get your example to work however and I am not sure why...
I am able to get it to work the way I think you intended by replacing the scalar product declarations like
```Mathematica
ScalarProduct[PolarizationVector[p1], PolarizationVector[p2]] = product12;
```
... etc, by explicit rules for the underlying representation of the scalar products in question:
```Mathematica
Pair[Momentum[Polarization[p1, I, Transversality -> True], D], Momentum[Polarization[p2, I, Transversality -> True], D]] =  product12;
Pair[Momentum[Polarization[p1, I, Transversality -> True], D], Momentum[Polarization[p3, -I, Transversality -> True], D]] = product13;
```
... etc (note the `-I` for complex conjugation on `p3` and `p4`). This solution is "brute force" and not preferable but at least it should enable you to get the results you were after until you find a proper solution. :slightly_smiling_face: 

\
_PS:_ this entire block:
```Mathematica
ScalarProduct[p1, p1] = SMP["m_Z"]^2;
ScalarProduct[p2, p2] = SMP["m_Z"]^2;
ScalarProduct[p3, p3] = SMP["m_Z"]^2;
ScalarProduct[p4, p4] = SMP["m_Z"]^2;
ScalarProduct[p1, p2] = (ss - 2 SMP["m_Z"]^2)/2;
ScalarProduct[p3, p4] = (ss - 2 SMP["m_Z"]^2)/2;
ScalarProduct[p1, p3] = -(tt - 2 SMP["m_Z"]^2)/2;
ScalarProduct[p2, p4] = -(tt - 2 SMP["m_Z"]^2)/2;
ScalarProduct[p1, p4] = -(uu - 2 SMP["m_Z"]^2)/2;
ScalarProduct[p2, p3] = -(uu - 2 SMP["m_Z"]^2)/2;
```
can be replaced by
```Mathematica
SetMandelstam[ss, tt, uu, p1, p2, -p3, -p4, SMP["m_Z"], SMP["m_Z"], SMP["m_Z"], SMP["m_Z"]];
```

**Reply by vsht** [reply]
With 
```
ScalarProduct[Polarization[p1, I, Transversality -> True], 
   Polarization[p2, I, Transversality -> True]] = product12;
ScalarProduct[Polarization[p1, I, Transversality -> True], 
  Polarization[p3, -I, Transversality -> True]] = product13; 
ScalarProduct[Polarization[p1, I, Transversality -> True], 
  Polarization[p4, -I, Transversality -> True]] = product14; 
ScalarProduct[Polarization[p2, I, Transversality -> True], 
  Polarization[p3, -I, Transversality -> True]] = product23;
ScalarProduct[Polarization[p2, I, Transversality -> True], 
   Polarization[p4, -I, Transversality -> True]] = product24;
ScalarProduct[Polarization[p3, -I, Transversality -> True], 
   Polarization[p4, -I, Transversality -> True]] = product34;
```
I get
```
(product12*product34*B0[ss, SMP["m_H"]^2, SMP["m_H"]^2]*SMP["e"]^4)/(128*Pi^2*SMP["cos_W"]^4*SMP["sin_W"]^4) + (product13*product24*B0[tt, SMP["m_H"]^2, SMP["m_H"]^2]*SMP["e"]^4)/(128*Pi^2*SMP["cos_W"]^4*SMP["sin_W"]^4) + 
 (product14*product23*B0[uu, SMP["m_H"]^2, SMP["m_H"]^2]*SMP["e"]^4)/(128*Pi^2*SMP["cos_W"]^4*SMP["sin_W"]^4)
```
Is this the expected output?

**Reply by roberpsm** [reply]
Thanks a lot for the replies, Magnus and Valdyslav. You suggestions are very useful and solve my problem!
